### PR TITLE
Remove: deprecated native MFA feature

### PR DESCRIPTION
**Pull request by peterhaochen47**
- Context about its deprecation:
  - This feature is under-utilized, and requires further maintenance for which our team lacks the resource. (For example, this feature is potentially vulnerable because a secure Content-Security-Policy cannot be applied to its pages without breaking them.) The feature has also been marked as "not ready for production" for a few years now. So we opt to remove the feature and instead recommend using the external IDPs's own MFA features. See more context in https://github.com/cloudfoundry/uaa/issues/2196.
- breaking changes planning: https://github.com/cloudfoundry/uaa-release/issues/739
- depends on: https://github.com/cloudfoundry/uaa-release/pull/784

 [#186854489]

### CODE REVIEW

**Comment by strehle** [comment]
@peterhaochen47 I think you can remove
libraries.zxing, https://github.com/cloudfoundry/uaa/blob/develop/dependencies.gradle#L129
libraries.googleAuth, https://github.com/cloudfoundry/uaa/blob/develop/dependencies.gradle#L55

**Review by hsinn0 — APPROVED** [review]

File: server/src/main/java/org/cloudfoundry/identity/uaa/audit/JdbcUnsuccessfulLoginCountingAuditService.java [hunk 1]
```diff
@@ -40,20 +39,11 @@ public JdbcUnsuccessfulLoginCountingAuditService(
...
-            case MfaAuthenticationSuccess:
-                resetAuthenticationEvents(auditEvent, zoneId, MfaAuthenticationFailure);
-
-                break;
             case UserAuthenticationSuccess:
-            case PasswordChangeSuccess:
-                resetAuthenticationEvents(auditEvent, zoneId, UserAuthenticationFailure);
-                break;
-            case UserAccountUnlockedEvent:
+            case PasswordChangeSuccess, UserAccountUnlockedEvent:
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/audit/JdbcUnsuccessfulLoginCountingAuditService.java]: see sonar, move this into one list

https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=2717&id=cloudfoundry-identity-parent&open=AY2GPGZ4RapHNV97UMiS

**Review by strehle — APPROVED** [review]
ok for us, see sonar comment

File: server/src/main/java/org/cloudfoundry/identity/uaa/audit/JdbcUnsuccessfulLoginCountingAuditService.java (hunk 1 above)
Comment (peterhaochen47) [reply server/src/main/java/org/cloudfoundry/identity/uaa/audit/JdbcUnsuccessfulLoginCountingAuditService.java]: Cool, fixed.

**Pull request merged**